src/java/org/apache/cassandra/service/accord/AccordStateCache.java

```diff
@@ -66,22 +60,27 @@ static class Stats
     private int unreferenced = 0;
     private long maxSizeInBytes;
     private long bytesCached = 0;
-    private final Stats stats = new Stats();
+    private final String metricsScope;
+
+    @VisibleForTesting
+    final AccordStateCacheMetrics metrics;
 
-    public AccordStateCache(ExecutorPlus loadExecutor, ExecutorPlus saveExecutor, long maxSizeInBytes)
+    public AccordStateCache(ExecutorPlus loadExecutor, ExecutorPlus saveExecutor, long maxSizeInBytes, String metricsScope)
     {
         this.loadExecutor = loadExecutor;
         this.saveExecutor = saveExecutor;
         this.maxSizeInBytes = maxSizeInBytes;
+        this.metricsScope = metricsScope;
+        this.metrics = new AccordStateCacheMetrics(String.format("%s-global", metricsScope), this);
```

Comment:
Assuming we only need global stats, or global stats and stats broken down by instance, would it be easier to just create 3 singleton instances of `AccordStateCacheMetrics` and reuse them across all command stores? I'm not sure whether this actually ends up constituting any correctness problems w/ multiple stores active (although I don't think we have any tests that cover this yet, as `AccordStateCacheTest` is more of a unit test), but it does seem like we would be doing a lot of duplicate MBean registration (or at least needless attempts at re-registration) in `registerMBean()`...

Reply:
Agreed

Reply:
hmm... it can be difficult because the `CacheMetrics` requires a reference to the `CacheSize` instance, a cache provider implemented by the cache itself. Although it seems justified to have a single metric for hits / misses / requests, regardless the number of cache instances, it does not apply to the cache size gauges, which we would have to sum up manually. If I understand that code correctly, we create only one cache instance per class per node

Reply:
`CacheSize` is just an interface though. We can still have our metrics collected in one object managed by `AccordCommandStores`, which can then delegate to the stores it manages to get an overall cache size. Check out `refreshCacheSizes()` where it interacts w/ the cache sizes already, albeit to set their size, not get it. Does that make more sense?